src/classes/CanvasRenderingContext2D.js

```diff
@@ -1282,22 +1271,26 @@ export default class CanvasRenderingContext2D {
 
   set shadowColor(value) {
     if (typeof value === 'string') {
-      const result = parseCSSColor(value);
-
-      if (result) {
-        this._shadowColorStack[this._stackIndex] = result;
-        const event = createCanvasEvent(
-          'shadowColor',
-          getTransformSlice(this),
-          { value: result },
-        );
-        this._events.push(event);
-      }
+      try { 
+        value = new MooColor(value);
+        this._shadowColorStack[this._stackIndex] = value;
```

Comment:
Should be

```ts
this._shadowColorStack[this._stackIndex] = value.toHex('short');
```

Reply:
I stand corrected. I am suggesting using a helper function that converts the `MooColor` to the correct string value used in the respective color stack.

```ts
// something like this?
function mooToColor(mooColorObject) {
  return  mooColorObject.getAlpha() === 1
    ? mooColorObject.toHex()
    : mooColorObject.toRGBA();
}
```

Reply:
Reasoning: We only need to parse the color once per set.